### PR TITLE
DEVDOCS-3567: [update] Add category examples to Stencil frontmatter's GraphQL reference

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-3567](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3567)

Added example for category on stencil doc's graphql section

[DEVDOCS-3567]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bc-andreadao** [comment]
Could use some help testing these. The graphQL code in the front matter block works on graphql playground when I tested it, but not sure about the handlebar example

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 1]
```diff
@@ -226,9 +226,15 @@ search:
 |`product_results`|`limit` defines the number of product search results displayed per page. The range of possible values is 1–100 products.|
   
 ## GraphQL attributes
-You can add [GraphQL Storefront API](/api-docs/storefront/graphql/graphql-storefront-api-overview) queries to your theme via the front matter block in a template file. For example, you can request a product's variants by augmenting the existing [product.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/product.html):
-  
- ```yaml lineNumbers
+You can add [GraphQL Storefront API](/api-docs/storefront/graphql/graphql-storefront-api-overview) queries to your theme via the front matter block in a template file. For example, you can request a product's variants by augmenting the existing [product.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/product.html), as well as a category's products by augmenting the existing [category.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/category.html):
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:229]:
```suggestion
You can add [GraphQL Storefront API](/api-docs/storefront/graphql/graphql-storefront-api-overview) queries to your theme using the front matter block in a template file. For example, you can request a product's variants by augmenting the existing [product.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/product.html), or a category's products by augmenting the existing [category.html template](https://github.com/bigcommerce/cornerstone/blob/master/templates/pages/category.html): 

```

remove via, change `as well as` to `or` to further support the narrative that these are examples & there are a lot more possibilities.

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 2]
```diff
@@ -255,12 +261,44 @@ gql: "query productById($productId: Int!) {
...
+type: tab
+title: Category
+-->
+
+ ```yaml title="Front matter block in category.html template" lineNumbers
+category:
+    shop_by_price: true
+    products:
+        limit: {{theme_settings.categorypage_products_per_page}}
+gql: "query categoryById ($categoryId: Int!){
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:275]: yeah, testing is a good idea. are we sure that the `gql` attribute is all the way left?

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 3]
```diff
@@ -255,12 +261,44 @@ gql: "query productById($productId: Int!) {
...
+        }
+      }
+    }
+  }
+}"
+```
+
+<!-- type: tab-end -->
+
 We suggest testing GraphQL queries using the [storefront API playground](https://developer.bigcommerce.com/graphql-playground) to refine them before adding them to your template. You can launch the playground in the context of your store by clicking the **Storefront API Playground** link under the **Advanced Settings** menu in your store's control panel.
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:292]:
```suggestion
We suggest testing and refining GraphQL queries before adding them to your template. You can use the [GraphQL Storefront API playground](https://developer.bigcommerce.com/graphql-playground). To launch the playground in the context of your store, navigate to the **Advanced Settings** menu in your store control panel and click the **Storefront API Playground** link.

```

@markcmurphy and I decided Friday that it's the GraphQL Storefront API, so made it say that. Also revised to leave more space for devs who have their own testing environments to drop out after we say "test it fam".

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 4]
```diff
@@ -255,12 +261,44 @@ gql: "query productById($productId: Int!) {
...
   
-Once you have added a query to your template's front matter block, execution happens automatically when the page loads. The data returned by the query will be returned in the page's context and made available to the handlebars under the `gql` key. For example, you can retrieve the variant data from the above query in `product.html` like this:
+Once you have added a query to your template's front matter block, execution happens automatically when the page loads. The data returned by the query will be returned in the page's context and made available to the handlebars under the `gql` key. The following examples illustrate how you can retrieve the variant data from the above query in `product.html`, as well as a category's product data from the above query in `category.html`:
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:294]: we can do less in this paragraph.

Handlebars is the proper name of a library, so it should always be capitalized. `the handlebars` should be Handlebars; when `Handlebars` appears following a definite article, Handlebars is an adjective modifying `object`, `implementation`, `context`, or similar. https://handlebarsjs.com/

```suggestion
Templates with queries defined in their front matter automatically execute their queries on page load. You can access the returned data using the `gql` Handlebars object in the page's context. The following examples illustrate how to access `gql` object data using results from the previously-defined front matter queries:

```

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 5]
```diff
@@ -255,12 +261,44 @@ gql: "query productById($productId: Int!) {
...
 
-```handlebars lineNumbers
+<!--
+type: tab
+title: Product
+-->
+
+```handlebars title="Handlebar in product.html template" lineNumbers
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:301]:
```suggestion
```handlebars title="Example gql Handlebars object in product.html template" lineNumbers

```

File: docs/stencil-docs/reference-docs/front-matter-reference.md [hunk 6]
```diff
@@ -269,6 +307,24 @@ Once you have added a query to your template's front matter block, execution hap
 {{/each}}
 {{/if}}
 ```
+
+<!--
+type: tab
+title: Category
+-->
+
+```handlebars title="Handlebar in category.html template" lineNumbers
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/front-matter-reference.md:316]:
```suggestion
```handlebars title="Example gql Handlebars object in category.html template" lineNumbers

```

**Review by slsriehl — CHANGES_REQUESTED** [review]
@bc-andreadao 

- I think @traci707 is the best person to help test the Stencil blocks. In addition to the actual objects, I am not sure the stuff inside your `{{# each}}` blocks is correctly scoped. Nested objects like this always need testing.
- Read through this article for other references to Handlebars and correct them per my comment. Alternatively, spin this bullet point off into another ticket.

**Comment by bc-andreadao** [comment]
converted to draft, working on creating better examples

**Comment by slsriehl** [comment]
Ticket paused.

**Comment by bc-andreadao** [comment]
Closing, Stencil not prioritized for 2024 roadmap

**Pull request closed without merging**